src/App.js

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect } from "react";
+import { getTodos, setSearchText, setTodosList } from "./actions";
+import { connect } from "react-redux";
+import "./App.css";
+import { DataTable, Paginator, SearchField } from "lucid-ui";
+import { actionTypes } from "./actions/actionTypes";
+
+const {
+  EmptyStateWrapper,
+  EmptyStateWrapper: { Title, Body },
+} = DataTable;
+
+const App = ({
+  fetchTodos,
+  setPage,
+  setPageSize,
+  setTodosList,
+  page,
+  todos,
+  page_size,
+  todos_list,
+  error,
+  search_text,
+  setSearch,
+  fetchSearchedTodos,
+}) => {
+  const columns = [
+    { field: "id", title: "ID", align: "left", width: 200 },
+    {
+      field: "title",
+      title: "TITLE",
+      align: "left",
+      width: 400,
+    },
+    {
+      field: "completed",
+      title: "COMPLETED",
+      align: "left",
+      width: 200,
+    },
+  ];
+  useEffect(() => {
+    fetchTodos();
+  }, [fetchTodos]);
+
+  useEffect(() => {
+    setTodosList(todos, page, page_size);
+  }, [page, page_size, todos]);
+
+  return (
+    <div className="App" data-test="App">
+      {error ? (
+        <span data-test="Error">Something Went Wrong!</span>
+      ) : (
+        <React.Fragment>
+          <div className="SearchField" data-test="SearchField">
+            <SearchField
+              placeholder={"Search by Title..."}
+              onChange={(value) => setSearch(value)}
+              value={search_text}
+              onSubmit={() => {
```

Comment:
extract all the logic in the onSubmit function to a separate function. It looks odd to put the logic here.